modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,305 @@
+<?php
+/**
+ * Module Name: Dominant Color
+ * Description: Adds support to store dominant-color for an image and create a placeholder background with that color.
+ * Experimental: Yes
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Add the dominant color metadata to the attachment.
+ *
+ * @since n.e.x.t
+ *
+ * @param array $metadata The attachment metadata.
+ * @param int   $attachment_id The attachment ID.
+ * @return array $metadata The attachment metadata.
+ */
+function dominant_color_metadata( $metadata, $attachment_id ) {
+	if ( ! wp_attachment_is_image( $attachment_id ) ) {
+		return $metadata;
+	}
+
+	$dominant_color = dominant_color_get_dominant_color( $attachment_id );
+
+	if ( is_string( $dominant_color ) && ! empty( $dominant_color ) ) {
+		$metadata['dominant_color'] = $dominant_color;
+	}
+
+	$has_transparency = dominant_color_get_has_transparency( $attachment_id );
+
+	if ( is_bool( $has_transparency ) ) {
+		$metadata['has_transparency'] = $has_transparency;
+	}
```

Comment:
Nit-pick, but a bit more "WordPress-y" would be to check if these are not `WP_Error`s.
```suggestion
	$dominant_color = dominant_color_get_dominant_color( $attachment_id );
	if ( ! is_wp_error( $dominant_color ) && ! empty( $dominant_color ) ) {
		$metadata['dominant_color'] = $dominant_color;
	}

	$has_transparency = dominant_color_get_has_transparency( $attachment_id );
	if ( ! is_wp_error( $has_transparency ) ) {
		$metadata['has_transparency'] = $has_transparency;
	}
```